### ISSUE TITLE
Example is failing for v0.9.22

### ISSUE THREAD

**Issue by xmnlab** [pdfmm/pdfmm#36]
Hi everyone! 

pdfmm looks a very interesting project! thanks for working on that!

as I mentioned in a previous issue, I am packaging pdfmm on conda-forge. 
Using that package I am not able to build the example at https://github.com/pdfmm/pdfmm/blob/v0.9.22/examples/helloworld/helloworld.cpp

This is the error that is rasing here::

```
bash ./scripts/optclean.sh
mkdir -p build
set -ex
meson setup \
	--prefix /home/xmn/mambaforge/envs/poc-pdfmm \
	--libdir /home/xmn/mambaforge/envs/poc-pdfmm/lib \
	--includedir /home/xmn/mambaforge/envs/poc-pdfmm/include \
	--includedir /home/xmn/mambaforge/envs/poc-pdfmm/include/pdfmm \
	--buildtype=release \
	--native-file meson.native  \
	build .
meson compile -C build
The Meson build system
Version: 0.63.2
Source dir: /mnt/sda1/storage/dev/inlyse-projects/poc-pdfmm
Build dir: /mnt/sda1/storage/dev/inlyse-projects/poc-pdfmm/build
Build type: native build
Project name: pocpdfmm
Project version: 1.0.0
C compiler for the host machine: /home/xmn/mambaforge/envs/poc-pdfmm/bin/x86_64-conda-linux-gnu-cc (gcc 10.4.0 "x86_64-conda-linux-gnu-cc (conda-forge gcc 10.4.0-16) 10.4.0")
C linker for the host machine: /home/xmn/mambaforge/envs/poc-pdfmm/bin/x86_64-conda-linux-gnu-cc ld.lld 14.0.4
C++ compiler for the host machine: /home/xmn/mambaforge/envs/poc-pdfmm/bin/x86_64-conda-linux-gnu-c++ (gcc 10.4.0 "x86_64-conda-linux-gnu-c++ (conda-forge gcc 10.4.0-16) 10.4.0")
C++ linker for the host machine: /home/xmn/mambaforge/envs/poc-pdfmm/bin/x86_64-conda-linux-gnu-c++ ld.lld 14.0.4
Host machine cpu family: x86_64
Host machine cpu: x86_64
Found pkg-config: /home/xmn/mambaforge/envs/poc-pdfmm/bin/pkg-config (0.29.2)
Run-time dependency cli11 found: YES 2.2.0
Found CMake: /home/xmn/mambaforge/envs/poc-pdfmm/bin/cmake (3.24.2)
Run-time dependency glog found: YES 0.6.0
WARNING: You should add the boolean check kwarg to the run_command call.
         It currently defaults to false,
         but it will default to true in future releases of meson.
         See also: https://github.com/mesonbuild/meson/issues/9300
Dependency gtest skipped: feature dev disabled
Dependency gmock skipped: feature dev disabled
Program clang-tidy skipped: feature dev disabled
Build targets in project: 3

pocpdfmm 1.0.0

  User defined options
    Native files: meson.native
    buildtype   : release
    includedir  : /home/xmn/mambaforge/envs/poc-pdfmm/include/pdfmm
    libdir      : /home/xmn/mambaforge/envs/poc-pdfmm/lib
    prefix      : /home/xmn/mambaforge/envs/poc-pdfmm

Found ninja-1.11.0 at /home/xmn/mambaforge/envs/poc-pdfmm/bin/ninja
ninja: Entering directory `/mnt/sda1/storage/dev/inlyse-projects/poc-pdfmm/build'
[0/1] Regenerating build files.
Cleaning... 0 files.
The Meson build system
Version: 0.63.2
Source dir: /mnt/sda1/storage/dev/inlyse-projects/poc-pdfmm
Build dir: /mnt/sda1/storage/dev/inlyse-projects/poc-pdfmm/build
Build type: native build
Project name: pocpdfmm
Project version: 1.0.0
C compiler for the host machine: /home/xmn/mambaforge/envs/poc-pdfmm/bin/x86_64-conda-linux-gnu-cc (gcc 10.4.0 "x86_64-conda-linux-gnu-cc (conda-forge gcc 10.4.0-16) 10.4.0")
C linker for the host machine: /home/xmn/mambaforge/envs/poc-pdfmm/bin/x86_64-conda-linux-gnu-cc ld.lld 14.0.4
C++ compiler for the host machine: /home/xmn/mambaforge/envs/poc-pdfmm/bin/x86_64-conda-linux-gnu-c++ (gcc 10.4.0 "x86_64-conda-linux-gnu-c++ (conda-forge gcc 10.4.0-16) 10.4.0")
C++ linker for the host machine: /home/xmn/mambaforge/envs/poc-pdfmm/bin/x86_64-conda-linux-gnu-c++ ld.lld 14.0.4
Host machine cpu family: x86_64
Host machine cpu: x86_64
Dependency CLI11 found: YES 2.2.0 (cached)
Dependency glog found: YES 0.6.0 (cached)
WARNING: You should add the boolean check kwarg to the run_command call.
         It currently defaults to false,
         but it will default to true in future releases of meson.
         See also: https://github.com/mesonbuild/meson/issues/9300
Dependency gtest skipped: feature dev disabled
Dependency gmock skipped: feature dev disabled
Program clang-tidy skipped: feature dev disabled
Build targets in project: 3

pocpdfmm 1.0.0

  User defined options
    Native files: /mnt/sda1/storage/dev/inlyse-projects/poc-pdfmm/meson.native
    backend     : ninja
    buildtype   : release
    includedir  : /home/xmn/mambaforge/envs/poc-pdfmm/include/pdfmm
    libdir      : /home/xmn/mambaforge/envs/poc-pdfmm/lib
    prefix      : /home/xmn/mambaforge/envs/poc-pdfmm

Found ninja-1.11.0 at /home/xmn/mambaforge/envs/poc-pdfmm/bin/ninja
[1/6] Linking static target liblibpdfmm.so.a
[2/6] Compiling C++ object libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o
[3/6] Linking target libpoc-pdfmm-lib.so
FAILED: libpoc-pdfmm-lib.so 
/home/xmn/mambaforge/envs/poc-pdfmm/bin/x86_64-conda-linux-gnu-c++  -o libpoc-pdfmm-lib.so libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o -L/home/xmn/mambaforge/envs/poc-pdfmm/lib -Wl,--as-needed -Wl,--no-undefined -fuse-ld=lld -Wl,-O1 -shared -fPIC -Wl,--start-group -Wl,-soname,libpoc-pdfmm-lib.so -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags -Wl,--gc-sections -Wl,--allow-shlib-undefined -Wl,-rpath,/home/xmn/mambaforge/envs/poc-pdfmm/lib -Wl,-rpath-link,/home/xmn/mambaforge/envs/poc-pdfmm/lib -fvisibility-inlines-hidden -std=c++17 -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /home/xmn/mambaforge/envs/poc-pdfmm/include -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem /home/xmn/mambaforge/envs/poc-pdfmm/include liblibpdfmm.so.a -lpthread /home/xmn/mambaforge/envs/poc-pdfmm/lib/libgflags.so.2.2.2 /home/xmn/mambaforge/envs/poc-pdfmm/lib/libglog.so.0.6.0 -Wl,--end-group
ld.lld: error: undefined symbol: vtable for mm::PdfString
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(mm::PdfString::~PdfString())
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(mm::PdfString::~PdfString())
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(HelloWorld(std::basic_string_view<char, std::char_traits<char> > const&))
>>> the vtable symbol may be undefined because the class is missing its key function (see https://lld.llvm.org/missingkeyfunction)

ld.lld: error: undefined symbol: mm::PdfDataProvider::~PdfDataProvider()
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(mm::PdfString::~PdfString())
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(mm::PdfString::~PdfString())
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(HelloWorld(std::basic_string_view<char, std::char_traits<char> > const&))
>>> referenced 4 more times

ld.lld: error: undefined symbol: mm::PdfError::PdfError(mm::PdfErrorCode, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(HelloWorld(std::basic_string_view<char, std::char_traits<char> > const&) (.cold))

ld.lld: error: undefined symbol: mm::PdfPainter::FinishDrawing()
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(HelloWorld(std::basic_string_view<char, std::char_traits<char> > const&) (.cold))
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(HelloWorld(std::basic_string_view<char, std::char_traits<char> > const&))

ld.lld: error: undefined symbol: mm::PdfEncoding::~PdfEncoding()
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(HelloWorld(std::basic_string_view<char, std::char_traits<char> > const&) (.cold))
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(HelloWorld(std::basic_string_view<char, std::char_traits<char> > const&))

ld.lld: error: undefined symbol: mm::PdfPainter::~PdfPainter()
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(HelloWorld(std::basic_string_view<char, std::char_traits<char> > const&) (.cold))
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(HelloWorld(std::basic_string_view<char, std::char_traits<char> > const&))

ld.lld: error: undefined symbol: mm::PdfMemDocument::~PdfMemDocument()
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(HelloWorld(std::basic_string_view<char, std::char_traits<char> > const&) (.cold))
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(HelloWorld(std::basic_string_view<char, std::char_traits<char> > const&))

ld.lld: error: undefined symbol: mm::PdfMemDocument::PdfMemDocument(bool)
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(HelloWorld(std::basic_string_view<char, std::char_traits<char> > const&))

ld.lld: error: undefined symbol: mm::PdfPainter::PdfPainter(mm::PdfPainterFlags)
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(HelloWorld(std::basic_string_view<char, std::char_traits<char> > const&))

ld.lld: error: undefined symbol: mm::PdfDocument::GetPages()
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(HelloWorld(std::basic_string_view<char, std::char_traits<char> > const&))

ld.lld: error: undefined symbol: mm::PdfPage::CreateStandardPageSize(mm::PdfPageSize, bool)
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(HelloWorld(std::basic_string_view<char, std::char_traits<char> > const&))

ld.lld: error: undefined symbol: mm::PdfPageCollection::CreatePage(mm::PdfRect const&)
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(HelloWorld(std::basic_string_view<char, std::char_traits<char> > const&))

ld.lld: error: undefined symbol: mm::PdfPainter::SetCanvas(mm::PdfCanvas*)
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(HelloWorld(std::basic_string_view<char, std::char_traits<char> > const&))

ld.lld: error: undefined symbol: mm::PdfEncoding::PdfEncoding()
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(HelloWorld(std::basic_string_view<char, std::char_traits<char> > const&))

ld.lld: error: undefined symbol: mm::PdfFontManager::GetFont(std::basic_string_view<char, std::char_traits<char> > const&, mm::PdfFontSearchParams const&, mm::PdfFontCreateParams const&)
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(HelloWorld(std::basic_string_view<char, std::char_traits<char> > const&))

ld.lld: error: undefined symbol: mm::PdfTextStateWrapper::SetFont(mm::PdfFont const*, double)
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(HelloWorld(std::basic_string_view<char, std::char_traits<char> > const&))

ld.lld: error: undefined symbol: mm::PdfPage::GetRect() const
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(HelloWorld(std::basic_string_view<char, std::char_traits<char> > const&))

ld.lld: error: undefined symbol: mm::PdfPainter::DrawText(std::basic_string_view<char, std::char_traits<char> > const&, double, double)
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(HelloWorld(std::basic_string_view<char, std::char_traits<char> > const&))

ld.lld: error: undefined symbol: mm::PdfDocument::GetInfo()
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(HelloWorld(std::basic_string_view<char, std::char_traits<char> > const&))
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(HelloWorld(std::basic_string_view<char, std::char_traits<char> > const&))
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(HelloWorld(std::basic_string_view<char, std::char_traits<char> > const&))
>>> referenced 2 more times

ld.lld: error: undefined symbol: mm::PdfString::PdfString(char const*)
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(HelloWorld(std::basic_string_view<char, std::char_traits<char> > const&))
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(HelloWorld(std::basic_string_view<char, std::char_traits<char> > const&))
>>> referenced by example.cpp
>>>               libpoc-pdfmm-lib.so.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_example.cpp.o:(HelloWorld(std::basic_string_view<char, std::char_traits<char> > const&))
>>> referenced 2 more times

ld.lld: error: too many errors emitted, stopping now (use -error-limit=0 to see all errors)
collect2: error: ld returned 1 exit status
[4/6] Compiling C++ object pocpdfmm.p/_mnt_sda1_storage_dev_inlyse-projects_poc-pdfmm_poc-pdfmm_src_main.cpp.o
/mnt/sda1/storage/dev/inlyse-projects/poc-pdfmm/poc-pdfmm/src/main.cpp: In function 'void show_version(int)':
/mnt/sda1/storage/dev/inlyse-projects/poc-pdfmm/poc-pdfmm/src/main.cpp:16:23: warning: unused parameter 'count' [-Wunused-parameter]
   16 | auto show_version(int count) -> void {
      |                   ~~~~^~~~~
ninja: build stopped: subcommand failed.
```

Another thing that I saw is that additionally to the include folder (that is really required), I need to specify "include/pdfmm" because in the headers files it is including "pdfmm_config.h". not sure if this is expected or if I am missing something. Normally I just need to point to my include folder from the (conda) environment and it is enough.

for instance, my code is here: https://github.com/xmnlab/poc-pdfmm
basically it just has infrastructure code (scripts, build system, etc) and the example from this repository. when I have it properly working, I will replace that by some code from my own.

**Comment by ceztko** [comment]
The include issue is possibly fixed in latest git, see https://github.com/pdfmm/pdfmm/commit/ff4bcd32521cc29cda79d4d23bbcbd761d416c02. I'm not sure about the undefined references: check which symbols are exported with [nm](https://stackoverflow.com/a/34796/213871) . The [way](https://github.com/conda-forge/staged-recipes/pull/20334/files) you are compiling pdfmm from a brief look it's ok to me.

**Comment by xmnlab** [comment]
thanks for your response @ceztko! I will keep the investigations here and I will update here when I find the solution :)
thanks!!

**Comment by xmnlab** [comment]
I managed to fixt he problem, I was an error in my meson build file. 
I changed the pdfmm dependency definition to 

```
deps = [
  dependency('CLI11'),
  dependency('glog'),
  cxx.find_library('pdfmm')
]
```

and it is now working. thanks!!
